docs/getting-started/dashboard_tutorial.md

```diff
@@ -0,0 +1,1788 @@
+                rx.text("Add User", size="4"),
+            ),
+        ),
+        rx.dialog.content(
+            rx.dialog.title(
+                "Add New User",
+            ),
+            rx.dialog.description(
+                "Fill the form with the user's info",
+            ),
+            rx.form(
+                rx.flex(
+                    rx.input(
+                        placeholder="User Name", name="name", required=True
+                    ),
+                    rx.input(
+                        placeholder="user@reflex.dev",
+                        name="email",
+                    ),
+                    rx.select(
+                        ["Male", "Female"],
+                        placeholder="Male",
+                        name="gender",
+                    ),
+                    rx.flex(
+                        rx.dialog.close(
+                            rx.button(
+                                "Cancel",
+                                variant="soft",
+                                color_scheme="gray",
+                            ),
+                        ),
+                        rx.dialog.close(
+                            rx.button(
+                                "Submit", type="submit"
+                            ),
+                        ),
+                        spacing="3",
+                        justify="end",
+                    ),
+                    direction="column",
+                    spacing="4",
+                ),
+                on_submit=State.add_user,
+                reset_on_submit=False,
+            ),
+            max_width="450px",
+        ),
+    )
+
+def index() -> rx.Component:
+    return rx.vstack(
+        add_customer_button(),
+        rx.table.root(
+            rx.table.header(
+                rx.table.row(
+                    rx.table.column_header_cell("Name"),
+                    rx.table.column_header_cell("Email"),
+                    rx.table.column_header_cell("Gender"),
+                ),
+            ),
+            rx.table.body(
+                rx.foreach(
+                    State.users, show_user
+                ),
+            ),
+            variant="surface",
+            size="3",
+        ),
+    )
+```
+
+
+## Plotting Data in a Graph
+
+The last part of this tutorial is to plot the user data in a graph. We will use Reflex's built-in graphing library recharts to plot the number of users of each gender. 
+
+### Transforming the data for the graph
+
+The graphing components in Reflex expect to take in a list of dictionaries. Each dictionary represents a data point on the graph and contains the x and y values. We will create a new event handler in the state called `transform_data` to transform the user data into the format that the graphing components expect. We must also create a new state variable called `users_for_graph` to store the transformed data, which will be used to render the graph.
```

Comment:
This is slightly confusing. So, is transform_data automatically run? I don't see you calling anywhere. Is it some kind of special thing that's particular to the graph component?

If so, mention that. And also how exactly it hooks up to it.

Reply:
added the info to the section after this

Reply:
yeah nice! I also think I just missed this function being called on my first readthrough. oops